### DISCUSSION TITLE
Custom Base URL

### DISCUSSION THREAD

**Discussion by 0xWOLAND** [vitejs/vite-plugin-react#317] [Q&A] [answered]
Hi all! I am trying to set the base URL for a ReactJS project to `qrc:///frontend/dist`, but vite doesn't seem to allow prefixes that aren't "https://" or "./". Please let me know if you have any suggestions! Thank you so much!

**Comment by ArnaudBarre** [comment]
Hey, you can open a feature request in Vite core with a title like `Support other protocols in base URL` explaining a bit more your setup and why you need both a custom protocol and a base path.

**Reply by 0xWOLAND** [reply]
Thank you! Do you have any ideas for workarounds for now?

**Reply by ArnaudBarre** [reply, marked as answer]
Yeah you can set the base path to `https://_to_be_replaced_` and then have a small post build steps where you read all JS files in the dist and to `content.replace("https://_to_be_replaced_", "qrc:///frontend/dist")`

**Reply by 0xWOLAND** [reply]
Do you know if there is a way to do this with Vite plugins?

**Reply by 0xWOLAND** [reply]
Figured it out :)

**Reply by ArnaudBarre** [reply]
This should do what you need:
```ts
{
      name: 'fix-base-path',
      renderChunk(code) {
        return code.replaceAll("https://_to_be_replaced_", "qrc:///frontend/dist")
      },
}
```
Maybe you need to check for index.html references too
